src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -91,3 +92,21 @@ def _parse_asn1_time(self, asn1_time):
             )
         ).decode("ascii")
         return datetime.datetime.strptime(time, "%Y%m%d%H%M%SZ")
+
+    @property
+    def signature_hash_algorithm(self):
+        # Set to 80 on the recommendation of
+        # https://www.openssl.org/docs/crypto/OBJ_nid2ln.html#return_values
+        buf_len = 80
+        buf = self._backend._ffi.new("char[]", buf_len)
+        res = self._backend._lib.OBJ_obj2txt(
+            buf, buf_len, self._x509.sig_alg.algorithm, 1
+        )
+        assert res <= 50 and res > 0
+        oid = self._backend._ffi.buffer(buf, res)[:].decode()
```

Comment:
50 would be because I missed it :) And yes, now that we're doing this in multiple places I'll rebase this against master once 1612 merges and fix this.